### ISSUE TITLE
Property of type Enum List fails to Map/Convert

### ISSUE THREAD

**Issue by mreyeros** [dotnet/efcore#18128]
The following simple model fails to map when querying with the Azure Cosmos Db provider:

```C#
public class FulfillmentOptionConfig {
        public string Id { get; set; }
        public FulfillmentOption FulfillmentOption { get; set; }
        public List<Peril> AllowedPerils { get; set; } = new List<Peril>();
        public string DocumentType { get; set; } = Constants.DocumentTypes.FulfillmentOptionConfig;
    }
```

I have attempted to convert the property using a ValueConverter but the following error is encountered:

Error: System.ArgumentException: Can not convert Array to String.
   at Newtonsoft.Json.Linq.JToken.op_Explicit(JToken value)
   at Newtonsoft.Json.Linq.JToken.ToObject(Type objectType)
   at Newtonsoft.Json.Linq.JToken.ToObject[T]()
   at lambda_method(Closure , QueryContext , JObject )
   at Microsoft.EntityFrameworkCore.Cosmos.Query.Internal.CosmosShapedQueryCompilingExpressionVisitor.AsyncQueryingEnumerable`1.AsyncEnumerator.MoveNextAsync()
   at Microsoft.EntityFrameworkCore.EntityFrameworkQueryableExtensions.ToListAsync[TSource](IQueryable`1 source, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.EntityFrameworkQueryableExtensions.ToListAsync[TSource](IQueryable`1 source, CancellationToken cancellationToken)
   at DF.ConfigManager.Features.Configuration.FulfillmentOptions.IndexMediator.QueryHandler.Handle(Query request, CancellationToken cancellationToken) in C:\dev\DF.ConfigManager\src\DF.ConfigManager\Features\Configuration\FulfillmentOptions\IndexMediator.cs:line 27
   at MediatR.Pipeline.RequestPostProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
   at MediatR.Pipeline.RequestPreProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
   at DF.ConfigManager.Features.Configuration.FulfillmentOptions.IndexBase.OnInitializedAsync() in C:\dev\DF.ConfigManager\src\DF.ConfigManager\Features\Configuration\FulfillmentOptions\Index.razor.cs:line 18
   at Microsoft.AspNetCore.Components.ComponentBase.RunInitAndSetParametersAsync()
   at Microsoft.AspNetCore.Components.RenderTree.Renderer.GetErrorHandledTask(Task taskToHandle)

If the property is ignored, the model is returned successfully.  I have another property, FulfillmentOption, which is of type enum and is setup to use the EnumToStringConverter and that is working without issue.

EF Core version: 3.0.0
Database provider: Microsoft.EntityFrameworkCore.Cosmos
Target framework: .NET Core 3.0
Operating system: Windows 10
IDE: Visual Studio 2019 16.3

**Comment by AndriySvyryd** [comment]
Duplicate of https://github.com/aspnet/EntityFrameworkCore/issues/14762

**Comment by ggirard07** [comment]
This issue has been closed as duplicate of #14762, but that one address primitive types like `string` or `int` but does not seem to support `enum`.
@AndriySvyryd is it possible to reopen?

**Comment by AndriySvyryd** [comment]
The new duplicate is https://github.com/dotnet/efcore/issues/25343
